cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -327,6 +327,11 @@ func (scaleSet *ScaleSet) TargetSize() (int, error) {
 
 // IncreaseSize increases Scale Set size
 func (scaleSet *ScaleSet) IncreaseSize(delta int) error {
+	if scaleSet.curSize == -1 {
+		klog.V(1).Infof("the scale set %s is under initialization, would skip the operation", scaleSet.Name)
+		return nil
```

Comment:
I think we should return an error still so autoscaler can backoff and try to resize a different agentpool.

In a super rare scenario where you have many nodepools - some of them are initialized and some aren't. You still want to to be able to scale whatever was initialized.